src/Perl6/bootstrap.c/BOOTSTRAP.nqp

```diff
@@ -592,10 +592,8 @@ my class Binder {
         # If it's attributive, now we assign it.
         if $is_attributive {
             # Find self.
-            my $self;
-            if nqp::existskey($lexpad, 'self') {
-                $self := nqp::atkey($lexpad, 'self');
-            } else {
+            my $self := nqp::atkey($lexpad, 'self');
```

Comment:
This will not work as intended because lexpads are not normal hashes. `atkey` on them throws `Lexical with name 'foo' does not exist in this frame` if a key is not found.